### DISCUSSION TITLE
Where is p2 profile used by Tycho cached?

### DISCUSSION THREAD

**Discussion by henryju** [eclipse-tycho/tycho#1305] [Q&A] [answered]
Hi,

(Sorry if I am not using the right names, I just discovered some of the concepts a few minutes ago.)

I was investigating a problem where our project was building fine on our CI platform, but we had strange compilation errors on our developer boxes. I say strange because the compiler was complaining that some types could not be resolved, while they were there in dependencies.

I was suspecting some cache issue around target platform resolution, because of the fact it was working fine on CI, and we have the same version of JDK/Maven on CI and local.

I tried removing `~.m2/repository`, since I know Tycho is caching things there, but still no success.

I ended up incrementing the `sequenceNumber` in our target file, and hourra, the compilation issue was gone.

I found in this [post](https://www.eclipse.org/forums/index.php/t/452909/) that:
> The sequence number is used to keep a target definition that uses p2 repositories in sync with a p2 profile. If something is changed to the target, the sequence number is incremented which indicates to the profile that it needs to catch up. The profile is used to cache the target content vastly improving performance.

So here is my question: where is the p2 profile used by Tycho cached?

**Comment by laeubi** [comment, marked as answer]
I'm not aware of any profile usage of Tycho, as you said, all data is stored inside `~.m2/repository` and/or `target` directories. Have you used `clean` goal? Also ejc has a problem with parallel execution so probably using `-T1` can help as well.

**Reply by henryju** [reply]
> Have you used `clean` goal?

Yes, but after ~double~ triple check, the target folder of one of the modules was not cleaned properly (inherited config from parent pom). So that was my issue. The fact it suddenly started to work after increasing the `sequenceNumber` was probably just a coincidence.

Thanks for your answer, it helped me to eliminate this wrong hypothesis.
